### ISSUE TITLE
validation time is too long

### ISSUE THREAD

**Issue by Yung-Chih-Lo** [THU-MIG/yolov10#203]
![image](https://github.com/THU-MIG/yolov10/assets/106985368/9ae00f25-5117-4110-a923-008ebb81fb7d)

Based on the picture shown below, the validation time seems excessively long. Is this normal for training?

CPU：AMD R9 3900
GPU：4070ti 12GB
RAM：32GB

task: detect
mode: train
model: yolov10b.pt
data: ../coco128.yaml
epochs: 500
time: null
patience: 100
batch: 12
imgsz: 640
save: true
save_period: -1
val_period: 1
cache: false
device: null
workers: 8
project: null
name: train
exist_ok: false
pretrained: true
optimizer: auto
verbose: true
seed: 0
deterministic: true
single_cls: false
rect: false
cos_lr: false
close_mosaic: 10
resume: false
amp: true
fraction: 1.0
profile: false
freeze: null
multi_scale: false
overlap_mask: true
mask_ratio: 4
dropout: 0.0
val: true
split: val
save_json: false
save_hybrid: false
conf: null
iou: 0.7
max_det: 300
half: false
dnn: false
plots: true
source: null
vid_stride: 1
stream_buffer: false
visualize: false
augment: false
agnostic_nms: false
classes: null
retina_masks: false
embed: null
show: false
save_frames: false
save_txt: false
save_conf: false
save_crop: false
show_labels: true
show_conf: true
show_boxes: true
line_width: null
format: torchscript
keras: false
optimize: false
int8: false
dynamic: false
simplify: false
opset: null
workspace: 4
nms: false
lr0: 0.01
lrf: 0.01
momentum: 0.937
weight_decay: 0.0005
warmup_epochs: 3.0
warmup_momentum: 0.8
warmup_bias_lr: 0.1
box: 7.5
cls: 0.5
dfl: 1.5
pose: 12.0
kobj: 1.0
label_smoothing: 0.0
nbs: 64
hsv_h: 0.015
hsv_s: 0.7
hsv_v: 0.4
degrees: 0.0
translate: 0.1
scale: 0.5
shear: 0.0
perspective: 0.0
flipud: 0.0
fliplr: 0.5
bgr: 0.0
mosaic: 1.0
mixup: 0.0
copy_paste: 0.0
auto_augment: randaugment
erasing: 0.4
crop_fraction: 1.0
cfg: null
tracker: botsort.yaml

**Comment by leonnil** [comment]
Thank you for your interest! It's not normal during training if you have much less images in validation set.

**Comment by Yung-Chih-Lo** [comment]
Thank you for your attention. So how can I fix this problem?

**Comment by leonnil** [comment]
Are you training on custom data? Please check that your validation path is correct and make sure you are validating on GPU device. Using `cache=True` and increasing your `batch` size might help.
